apps/catalyst-ui-worker/components/modals/CreateChannelModal.tsx

```diff
@@ -0,0 +1,181 @@
+'use client';
+import { useState, useEffect } from 'react';
+import { useRouter } from 'next/navigation';
+import {
+    FormControl,
+    FormLabel,
+    Grid,
+    Input,
+    Modal,
+    ModalBody,
+    ModalCloseButton,
+    ModalContent,
+    ModalFooter,
+    ModalHeader,
+    ModalOverlay,
+    Text,
+    Flex,
+    FormErrorMessage,
+    useDisclosure,
+} from '@chakra-ui/react';
+import { OrbisButton } from '@/components/elements';
+import { createDataChannel } from '@/app/actions/channels';
+import { DataChannel } from '@catalyst/schemas';
+
+type CreateChannelModalProps = {
+    disclosure: ReturnType<typeof useDisclosure>;
+    user?: { custom: { org?: string } };
+    token?: string;
+};
+
+export function CreateChannelModal({ disclosure, user, token }: CreateChannelModalProps) {
+    const router = useRouter();
+    const [newChannel, setNewChannel] = useState<Omit<DataChannel, 'id'>>({
+        name: '',
+        description: '',
+        endpoint: '',
+        creatorOrganization: String(user?.custom.org || ''),
+        accessSwitch: true,
+    });
+    const [nameError, setNameError] = useState<string>('');
+    const [isSubmitting, setIsSubmitting] = useState<boolean>(false);
+
+    // Update creatorOrganization when user changes
+    useEffect(() => {
+        if (user?.custom.org) {
+            setNewChannel((prev) => ({
+                ...prev,
+                creatorOrganization: String(user.custom.org),
+            }));
+        }
+    }, [user?.custom.org]);
+
+    const resetForm = () => {
+        setNewChannel({
+            name: '',
+            description: '',
+            endpoint: '',
+            creatorOrganization: String(user?.custom.org || ''),
+            accessSwitch: true,
+        });
+        setNameError('');
+    };
+
+    const handleSubmit = (formData: FormData) => {
+        setIsSubmitting(true);
+        setNameError('');
+        formData.set('organization', String(user?.custom.org));
+
+        if (!token) {
+            setNameError('Authentication required');
+            setIsSubmitting(false);
+            return;
+        }
+
+        createDataChannel(formData, token)
+            .then((result) => {
+                if (result.success && result.data) {
+                    disclosure.onClose();
+                    resetForm();
+                    router.push('/channels/' + result.data.id);
```

Comment:
@jaeyojae Here, `isSubmitting` is never set back to false. Because the component stays mounted, reopening the modal leaves all inputs disabled and the submit button permanently loading until the page is refreshed

```suggestion
                    disclosure.onClose();
                    resetForm();
                    setIsSubmitting(false);
                    router.push('/channels/' + result.data.id);
```